sdks/python/src/opik/api_objects/opik_client.py

```diff
@@ -437,6 +437,107 @@ def get_dataset(self, name: str) -> dataset.Dataset:
 
         return dataset_
 
+    def get_datasets(
+        self,
+        max_results: int = 100,
+        sync_items: bool = True,
+    ) -> List[dataset.Dataset]:
+        """
+        Returns all datasets up to the specified limit.
+
+        Args:
+            max_results: The maximum number of datasets to return.
+            sync_items: Whether to sync the hashes of the dataset items. This is used to deduplicate items when fetching the dataset but it can be an expensive operation.
+
+        Returns:
+            List[dataset.Dataset]: A list of dataset objects that match the filter string.
+        """
+        page_size = 100
```

Comment:
We should lighten the opik_client content and reduce the logic stored directly in its methods.
For `get_datasets` function let's create a module `opik.api_objects.dataset.rest_operations`.
And put this function here:
```python
def get_datasets(rest_client, max_results, sync_items) -> List[dataset.Dataset]:
   ...
```